reactor-aeron-core/src/main/java/reactor/aeron/AeronEventLoop.java

```diff
@@ -0,0 +1,213 @@
+
+  // Worker
+  private final Mono<Worker> workerMono;
+  private volatile Thread thread;
+
+  // Commands
+  private final Queue<CommandTask> commandTasks = new ConcurrentLinkedQueue<>();
+
+  // Senders
+  private final List<MessagePublication> publications = new ArrayList<>();
+
+  AeronEventLoop() {
+    workerMono = Mono.fromCallable(this::createWorker).cache();
+  }
+
+  private Worker createWorker() {
+    ThreadFactory threadFactory = defaultThreadFactory();
+    Worker w = new Worker();
+    thread = threadFactory.newThread(w);
+    thread.start();
+    return w;
+  }
+
+  boolean inEventLoop() {
+    return thread == Thread.currentThread();
+  }
+
+  public Mono<Void> execute(Consumer<MonoSink<Void>> consumer) {
+    return worker().flatMap(w -> command(consumer));
+  }
+
+  public Mono<Void> register(MessagePublication publication) {
+    return worker().flatMap(w -> command(sink -> register(publication, sink)));
+  }
+
+  private void register(MessagePublication publication, MonoSink<Void> sink) {
+    Objects.requireNonNull(publication, "messagePublication must be not null");
+    publications.add(publication);
+    sink.success();
+  }
+
+  public Mono<Void> dispose(MessagePublication publication) {
+    return worker().flatMap(w -> command(sink -> dispose(publication, sink)));
+  }
+
+  @Override
+  public void dispose() {
+    dispose.onComplete();
+  }
+
+  private void dispose(MessagePublication publication, MonoSink<Void> sink) {
+    publications.removeIf(p -> p == publication);
+    Optional.ofNullable(publication).ifPresent(MessagePublication::close);
+    sink.success();
+  }
+
+  @Override
+  public Mono<Void> onDispose() {
+    return onDispose;
+  }
+
+  @Override
+  public boolean isDisposed() {
+    return onDispose.isDisposed();
+  }
+
+  private static ThreadFactory defaultThreadFactory() {
+    return r -> {
+      Thread thread = new Thread(r);
+      thread.setName("aeron-event-loop");
+      thread.setUncaughtExceptionHandler(
+          (t, e) -> logger.error("Uncaught exception occurred: {}", e, e));
+      return thread;
+    };
+  }
+
+  private Mono<Worker> worker() {
+    return workerMono.takeUntilOther(listenDispose());
+  }
+
+  private Mono<Void> command(Consumer<MonoSink<Void>> consumer) {
+    return Mono.create(sink -> commandTasks.add(new CommandTask(sink, consumer)));
+  }
+
+  private <T> Mono<T> listenDispose() {
+    //noinspection unchecked
+    return dispose //
+        .map(avoid -> (T) avoid)
+        .switchIfEmpty(Mono.error(Exceptions::failWithRejected));
+  }
+
+  /**
+   * Runnable event loop worker. Does a following:
+   *
+   * <ul>
+   *   <li>runs until dispose signal obtained
+   *   <li>on run iteration makes progress on: a) commands; b) publications; c) subscriptions
+   *   <li>idles on negative progress
+   * </ul>
+   */
+  private class Worker implements Runnable {
+
+    private final IdleStrategy idleStrategy = AeronUtils.newBackoffIdleStrategy();
```

Comment:
this can be changed somehow to spinning if user decides?

Reply:
Changed that. Now u can provide in AeronResourcesConfig  idleStrategy